SimCore/src/SimCore/Generators/LHEPrimaryGenerator.cxx

```diff
@@ -57,48 +43,47 @@ void LHEPrimaryGenerator::GeneratePrimaryVertex(G4Event* anEvent) {
                             "Failed to find particle definition for W ion.");
           }
         } else {
-          primary->SetPDGcode(particle->getIDUP());
+          primary->SetPDGcode(particle->getPdgId());
         }
 
+        // TODO: should we not require that the primary is an electron?
+        // for example the WAB LHE events will have the wide photon as primary
+
         primary->Set4Momentum(
-            particle->getPUP(0) * GeV, particle->getPUP(1) * GeV,
-            particle->getPUP(2) * GeV, particle->getPUP(3) * GeV);
-        primary->SetProperTime(particle->getVTIMUP() * nanosecond);
+            particle->getMomentum(0) * GeV, particle->getMomentum(1) * GeV,
+            particle->getMomentum(2) * GeV, particle->getMomentum(3) * GeV);
+        primary->SetProperTime(particle->getLifetime() * nanosecond);
 
-        UserPrimaryParticleInformation* primaryInfo =
-            new UserPrimaryParticleInformation();
-        primaryInfo->setHepEvtStatus(particle->getISTUP());
-        primary->SetUserInformation(primaryInfo);
+        auto primary_info = std::make_unique<UserPrimaryParticleInformation>();
+        primary_info->setHepEvtStatus(particle->getStatus());
+        primary->SetUserInformation(primary_info.release());
 
-        particleMap[particle] = primary;
+        particleMap[particle.get()] = primary;
 
         /*
          * Assign primary as daughter but only if the mother is not a DOC
-         * particle.
+         * particle->
          */
-        if (particle->getMother(0) != nullptr &&
-            particle->getMother(0)->getISTUP() > 0) {
-          G4PrimaryParticle* primaryMom = particleMap[particle->getMother(0)];
-          if (primaryMom != nullptr) {
-            primaryMom->SetDaughter(primary);
+        if (particle->getMotherParticle(0) != nullptr &&
+            particle->getMotherParticle(0)->getStatus() > 0) {
+          G4PrimaryParticle* primary_mom =
+              particleMap[particle->getMotherParticle(0)];
+          if (primary_mom != nullptr) {
+            primary_mom->SetDaughter(primary);
           }
         } else {
           vertex->SetPrimary(primary);
         }
-      }
+      }  // end condition for valid, outgoing particle status
     }
 
     anEvent->AddPrimaryVertex(vertex);
 
   } else {
-    std::cout << "[ LHEPrimaryGenerator ] : Ran out of input events so run "
-                 "will be aborted!"
-              << std::endl;
+    ldmx_log(info) << "Ran out of input events so run will be aborted!";
```

Comment:
I would consider this an `error` since the user should not ask for more events than they are providing with the LHE file.

```suggestion
    ldmx_log(error) << "Ran out of input events so run will be aborted!";
```